internal/declarative/v2/reconciler.go

```diff
@@ -198,23 +185,25 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		if errors.Is(err, ErrClientUnauthorized) {
 			r.invalidateClientCache(ctx, obj)
 		}
-		return r.ssaStatusIfDiffExist(ctx, obj, metrics.ManifestSyncResources, currentObjStatus, err)
+		return r.dealWithReconcileError(ctx, req.Name, obj, metrics.ManifestSyncResources, currentObjStatus, err)
 	}
 
 	// This situation happens when manifest get new installation layer to update resources,
 	// we need to make sure all updates successfully before we can update synced oci ref
 	if requireUpdateSyncedOCIRefAnnotation(obj, spec.OCIRef) {
 		updateSyncedOCIRefAnnotation(obj, spec.OCIRef)
-		return r.updateObject(ctx, obj, metrics.ManifestUpdateSyncedOCIRef, queue.IntendedRequeue)
+		return r.updateObject(ctx, obj, metrics.ManifestUpdateSyncedOCIRef)
 	}
 
 	if !obj.GetDeletionTimestamp().IsZero() {
 		r.cleanUpMandatoryModuleMetrics(obj)
 		r.ManifestMetrics.RemoveManifestDuration(req.Name)
-		return r.removeFinalizers(ctx, obj, []string{r.Finalizer}, metrics.ManifestRemoveFinalizerInDeleting, nil)
```

Comment:
To reviewer:
previous e2e failing is due to this line, the error pass to `removeFinalizers` is `nil, which leads to requeue without error.
I refactored the `removeFinalizers` to not take care requeue logic.